### DISCUSSION TITLE
I can't figure out how to use the Sprite Previewer, can someone help me

### DISCUSSION THREAD

**Discussion by BestKitsune** [Monika-After-Story/MonikaModDev#8227] [QA - code and content creation (technical)]
Can someone give me a very detailed how-to-get the Sprite Previewer to work. If you could that would be great.

Also I am trying to learn to code. Can someone give me a good place to learn python. It would be great if the program is free. Thanks for maybe listening.

**Comment by ThePotatoGuy** [comment]
for the sprite previewer, see [this faq entry](https://github.com/Monika-After-Story/MonikaModDev/wiki/FAQ#how-do-i-find-the-spritecode-for-an-expression)

For python, im sure code academy is ok

**Reply by BestKitsune** [reply]
Okay so I did what it told me to do and put it in the game folder and it would no longer open. Why did it do that?

**Reply by ThePotatoGuy** [reply]
Is there a traceback or anything? What happens when you launch the mod?

**Reply by BestKitsune** [reply]
okay I got it to work

**Reply by WishStone-Dragon** [reply]
I wanted to add in I had the same problem with the game no longer opening. 
I literally do not know what I am doing, so chances are the problem lies therein. What I did was:
I made a copy of the DDLC folder + MAS and stuck it into its own area called _C:\MAS Sandbox\Doki Doki Literature Club Expressions_. I was hoping I could somehow create a safe area that wouldn't impact the main game.

I also attempted to combine this with the "[Bug fixes and new software features](https://github.com/Monika-After-Story/MonikaModDev/wiki/Contributing-Guidelines#bug-fixes-and-new-software-features)" information given to create a localized persistent by editing DDLC.py as instructed. However, without a picture or similar to go by, I am flying blind. This is what I ended up with after following the description:
![image](https://user-images.githubusercontent.com/95390551/147620161-ed48bccc-ed5a-4536-8ef3-2c41c910944c.png)

What happens is: If I double-click the DDLC.exe in that separate folder with that edited ddlc.py, is nothing I can see. 

I tried to run as admin with the same result.

I tried to launch the ddlc.exe with the task manager open. It seems to not even register the opening of the exe.

The only traceback.txt is the fake one Monika writes in the game

There's nothing I can see in the log folder or in any of the files therein that points to an issue.

**Reply by Booplicate** [reply]
You have incorrect indentation. If you try use a more advanced code editor, it'll show you an error.
```py
    if save_directory is None:
        return gamedir + "/saves"
```

**Reply by WishStone-Dragon** [reply]
Ah, guess Notepad++ couldn't get me that kind of detail. Not a coder, so I am fumbling in the dark. x3

THAT launches. Good grief, I cannot wait to share this with the friend who stared at this with me for 30 minutes the other day. Thank you <3
